internal/runtime/internal/controller/component_registry.go

```diff
@@ -16,13 +16,15 @@ type ComponentRegistry interface {
 
 type defaultComponentRegistry struct {
 	minStability featuregate.Stability
+	community    bool
 }
 
 // NewDefaultComponentRegistry creates a new [ComponentRegistry] which gets
 // components registered to github.com/grafana/alloy/internal/component.
-func NewDefaultComponentRegistry(minStability featuregate.Stability) ComponentRegistry {
+func NewDefaultComponentRegistry(minStability featuregate.Stability, community bool) ComponentRegistry {
```

Comment:
I don't think this is a good name. Does `community` mean that this registry is exclusively for community components? Does it mean that it's a community stability level? I think the naming should be more clear.